packages/components/src/internal/announcements/ThreadEditor.tsx

```diff
@@ -417,10 +420,16 @@ export const ThreadEditor: FC<ThreadEditorProps> = props => {
         (event: ChangeEvent<HTMLTextAreaElement>) => {
             setError(undefined);
             setBody(event.target.value);
+            setPendingChange?.(thread?.rowId ?? -1, !!event.target.value);
```

Comment:
nit: The props to `setPendingChange` seem to be consistent. Consider consolidating into a handler:

```tsx
const handlePendingChange = useCallback((hasPendingChange = false) => {
    setPendingChange?.(thread?.rowId ?? -1, hasPendingChange);
}, [setPendingChange, thread?.rowId]);

const onBodyChange = useCallback(
    (event: ChangeEvent<HTMLTextAreaElement>) => {
        setError(undefined);
        setBody(event.target.value);
        handlePendingChange(!!event.target.value);
    },
    [handlePendingChange, setBody]
);

const handleCancel = useCallback(() => {
    onCancel?.();
    handlePendingChange();
}, [handlePendingChange, onCancel, setPendingChange]);

const onSubmit = useCallback(() => {
    if (submitting) return;
    setSubmitting(true);
    if (isCreate) {
        createThread();
    } else {
        updateThread();
    }
    handlePendingChange();
}, [createThread, handlePendingChange, isCreate, setSubmitting, submitting, updateThread]);
```